File: languages/php/Bloock/Entity/Record/AuthenticityDetails.php
<?php

namespace Bloock\Entity\Record;

use Bloock\Entity\Authenticity\Signature;

/**
 * Represents details related to the authenticity of a record, including signatures.
 */
class AuthenticityDetails
{
    private array $signatures;

    /**
     * Constructs a AuthenticityDetails object with the specified parameters.
     * @param array $signatures
     */
    public function __construct(array $signatures)
    {
        $this->signatures = $signatures;
    }

    /**
     * Gets the signatures of the record.
     * @return array
     */
    public function getSignatures(): array
    {
        return $this->signatures;
    }

    public static function fromProto(\Bloock\AuthenticityDetails $details): AuthenticityDetails
    {
        $signatures = [];
        foreach ($details->getSignatures() as $signature) {
            $signatures[] = Signature::fromProto($signature);
        }
        return new AuthenticityDetails($signatures);
    }

    public function toProto(): \Bloock\AuthenticityDetails
    {
        $p = new \Bloock\AuthenticityDetails();

        $signatures = [];
        foreach ($this->signatures as $signature) {
            $signatures[] = $signature->toProto();
        }
        $p->setSignatures($signatures);
        return $p;
    }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/record/AuthenticityDetails.java
package com.bloock.sdk.entity.record;

import com.bloock.sdk.bridge.proto.BloockRecordEntities;
import com.bloock.sdk.entity.authenticity.Signature;
import java.util.List;
import java.util.stream.Collectors;

/**
 * Represents details related to the authenticity of a record, including
 * signatures.
 */
public class AuthenticityDetails {
  List<Signature> signatures;

  /**
   * Constructs a AuthenticityDetails object with the specified parameters.
   * 
   * @param signatures
   */
  AuthenticityDetails(List<Signature> signatures) {
    this.signatures = signatures;
  }

  /**
   * Gets the signatures of the record.
   * 
   * @return
   */
  public List<Signature> getSignatures() {
    return signatures;
  }

  public static AuthenticityDetails fromProto(BloockRecordEntities.AuthenticityDetails details) {
    List<Signature> signatures = details.getSignaturesList().stream().map(Signature::fromProto)
        .collect(Collectors.toList());
    return new AuthenticityDetails(signatures);
  }

  public BloockRecordEntities.AuthenticityDetails toProto() {
    return BloockRecordEntities.AuthenticityDetails.newBuilder()
        .addAllSignatures(signatures.stream().map(Signature::toProto).collect(Collectors.toList()))
        .build();
  }
}
